javascript/ql/src/semmle/javascript/frameworks/Koa.qll

```diff
@@ -100,6 +122,77 @@ module Koa {
     }
   }
 
+  /**
+   * A Koa route handler registered using a routing library.
+   *
+   * Example of what that could look like:
+   * ```JavaScript
+   * const router = require('koa-router')();
+   * const Koa = require('koa');
+   * const app = new Koa();
+   * router.get('/', async (ctx, next) => {
+   *    // route handler stuff
+   * });
+   * app.use(router.routes());
+   * ```
+   */
+  private class RoutedRouteHandler extends RouteHandler, DataFlow::SourceNode {
+    DataFlow::InvokeNode router;
+    DataFlow::MethodCallNode call;
+
+    RoutedRouteHandler() {
+      router = DataFlow::moduleImport(["@koa/router", "koa-router"]).getAnInvocation() and
+      call = router.getAMethodCall*() and
+      call.getMethodName() =
```

Comment:
We could use `getAChainedMethodCall` here, but I'll leave it up to you if you want to use it. It's unclear to me what the performance impact would be.

Reply:
Thanks, I had forgotten about `getAChainedMethodCall`. 

I think the performance impact is negligible, it will likely unfold to the same thing anyway.   
I can do another smoke-test evaluation if you like.